### PR TITLE
FM 156 - Implement Handlers

### PR DESCRIPTION
**Pull request by milankarunarathne**
Created File Structure for Diagnostic Report
    
      * Added files to handle server request with
        - RestfulDiagnosticReportResourceProvider.java
        - FHIRDiagnosticReportResource.java
      * Do Diagnostic Report mapping with OpenMRS API
        - DiagnosticReportService.java
        - DiagnosticReportServiceImpl.java
      * Implement Handler with
        - FHIRDiagnosticReportUtil.java

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir/api/DiagnosticReportService.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	 */
+	DiagnosticReport getDiagnosticReport(String id);
+	
+	/**
+	 * Creates OpenMRS objects from FHIR Diagnostic Report
+	 *
+	 * @param diagReport
+	 * @return
+	 */
+	DiagnosticReport createFHIRDiagnosticReport(DiagnosticReport diagReport);
```
Comment (surangak) [inline api/src/main/java/org/openmrs/module/fhir/api/DiagnosticReportService.java]: Use full variable names "diagnosticReport"

File: api/src/main/java/org/openmrs/module/fhir/api/LocationService.java [hunk 2]
```diff
@@ -68,5 +68,12 @@
 	 * @param location representation of location fhir resource
 	 */
 	public void updateLocationById(String id, Location location);
+	
+	/**
+	 * Create location 
+	 *
+	 * @param location representation of location fhir resource
+	 */
+	public Location createLocation(Location location);
```
Comment (surangak) [inline api/src/main/java/org/openmrs/module/fhir/api/LocationService.java]: Hm..... I see why you need this....

Comment (milankarunarathne) [inline api/src/main/java/org/openmrs/module/fhir/api/LocationService.java]: No no, those are not mine. I get a pull --rebase and Sashrika's commits append top of my commits.

Comment (milankarunarathne) [inline api/src/main/java/org/openmrs/module/fhir/api/LocationService.java]: Can you see https://goo.gl/tNiSpK, I'm not there.

File: api/src/main/java/org/openmrs/module/fhir/api/DiagnosticReportService.java (hunk 1 above)
Comment (milankarunarathne) [inline api/src/main/java/org/openmrs/module/fhir/api/DiagnosticReportService.java]: Fixed and push the code.

**Comment by harsha89** [comment]
@milankarunarathne this pull request has messed up several things. It's also contained sashirkas changes as well.

When you using git, often update your repo with remote. Use git pull --rebase upstream master to get changes from remote. Can you execute git pull again and merge changes and clean the pull request?

**Comment by milankarunarathne** [comment]
@harsha89 I did as you said. Please, check now.
`git pull --rebase upstream master`
`git pull --rebase origin FM-156` to fix issues
` git push origin FM-156`
then it says `Everything up-to-date`

**Pull request merged**